ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7852,6 +7852,10 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
           throw new SemanticException("Error while getting the full qualified path for the given directory: " + ex.getMessage());
         }
       }
+
+      if (!isNonNativeTable && AcidUtils.isTransactionalTable(destinationTable) && qb.isCTAS()) {
```

Comment:
is `!isNonNativeTable` required to exclude Iceberg tables?

Reply:
Yes

Reply:
is there a reason, why we should exclude them? additional Iceberg-related changes are required?

Reply:
I enquired with @lcspinter offline, Iceberg tables are treated as external tables and specifically for Iceberg tables, there is a implementation of cleanup here - 
https://github.com/apache/hive/blob/master/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergQueryLifeTimeHook.java